lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -3617,12 +3617,12 @@ void Solution::dump(raw_ostream &out) const {
   }
 }
 
-void ConstraintSystem::dump() {
-  print(llvm::errs());
+void ConstraintSystem::dump() const {
+  const_cast<ConstraintSystem*>(this)->print(llvm::errs());
```

Comment:
I think you can drop these by marking `ConstraintSystem::print` const.

Reply:
Took a bit of doing, but I did it. I've also squashed the initial commits.

Reply:
Took a bit of doing, but I did it. I've also squashed the initial commits.